Neos.Flow/Classes/Mvc/Routing/Dto/ResolveContext.php

```diff
@@ -50,18 +50,25 @@ final class ResolveContext
      */
     private $uriPathPrefix;
 
+    /**
+     * @var RouteParameters
+     */
+    private $parameters;
+
     /**
      * @param UriInterface $baseUri The base URI, retrieved from the current request URI or from configuration, if specified. Required to fill in parts of the result when resolving absolute URIs
      * @param array $routeValues Route values to build the URI, for example ['@action' => 'index', 'someArgument' => 'foo', ...]
      * @param bool $forceAbsoluteUri Whether or not an absolute URI is to be returned
      * @param string $uriPathPrefix A prefix to be prepended to any resolved URI. Not allowed to start with "/".
+     * @param RouteParameters $parameters
      */
-    public function __construct(UriInterface $baseUri, array $routeValues, bool $forceAbsoluteUri, string $uriPathPrefix = '')
+    public function __construct(UriInterface $baseUri, array $routeValues, bool $forceAbsoluteUri, string $uriPathPrefix, RouteParameters $parameters)
```

Comment:
I somehow think the last three arguments should be optional with sane defaults (absoluteUri false, uriPathPrefix '' and empty parameters)

Reply:
Mathias Vereas once criticised the many optional arguments in our code and I didn't really agree at first. But in the meantime I learned to hate them too. In this case the invoking side should make an informed decision about whether to force absolute URIs etc. In practice this code will probably never be invoked in user land code anyways